template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/AbstractRoleConfigProvider.java

```diff
@@ -30,7 +31,7 @@
 
         Map<String, List<ApiClusterTemplateConfig>> configs = new HashMap<>();
 
-        for (ApiClusterTemplateRoleConfigGroup rcg : ofNullable(service.get().getRoleConfigGroups()).orElseGet(List::of)) {
+        for (ApiClusterTemplateRoleConfigGroup rcg : ofNullable(service.get().getRoleConfigGroups()).orElse(new ArrayList<>())) {
             if (getRoleTypes().contains(rcg.getRoleType())) {
```

Comment:
While the change does not hurt, I think it is not a fix for the original problem. Quoting the stack trace from the Jira ticket:
```
Caused by: java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.ImmutableCollections$AbstractImmutableList.indexOf(ImmutableCollections.java:170)
	at java.base/java.util.ImmutableCollections$AbstractImmutableList.contains(ImmutableCollections.java:201)
	at com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigProvider.getRoleConfigs(AbstractRoleConfigProvider.java:34)
```
The problem originates at line 34, which is the `if ()` inside the `for` loop. The NPE is raised when `rcg.getRoleType() == null`.